### PR TITLE
Update Authentication and documentation

### PR DESCRIPTION
**Pull request by BrettGAxosoft**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+
+This library is built to support v6 of the Axosoft REST API and has been tested with both axosoft.com hosted accounts and Axosoft installed on premise.  Any questions can be sent to api@axosoft.com.
+
+[Here's](http://developer.axosoft.com/) a link to our full documentation for the Axosoft REST API.
+
+## Installation
+
+  Install with the node package manager [npm](http://npmjs.org):
+
+    $ npm install node-axosoft
```
Comment (jdgarcia) [inline README.md]: I'd add a `--save` to this. Most package READMEs (or at least the one's I've seen) usually include it.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,81 @@
...
+
+### Create the Axosoft Connection
+
+    const nodeAxosoft = require('node-axosoft');
+    const axosoftUrl = {Axosoft Url};
+    var credentials = {};
+
+    //Populate Credentials (See Below)
+
+    var axo = new nodeAxosoft(axosoftUrl, credentials);
```
Comment (jdgarcia) [inline README.md]: You should consistently use either `const` or `var`.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,81 @@
...
+  Install with the node package manager [npm](http://npmjs.org):
+
+    $ npm install node-axosoft
+
+## Examples
+
+### Create the Axosoft Connection
+
+    const nodeAxosoft = require('node-axosoft');
+    const axosoftUrl = {Axosoft Url};
```
Comment (jdgarcia) [inline README.md]: I understand what this is but it looks a bit awkward with an actual object declaration right below it. Maybe just `https://example.axosoft.com/`?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,81 @@
...
+
+### Populate Credentials
+
+  User Name and Password
+
+    credentials.client_id = {client id};
+    credentials.client_secret = {client_secret};
+    credentials.grant_type = 'password';
+    credentials.username = {username};
+    credentials.password = {password};
```
Comment (jdgarcia) [inline README.md]: same with these

File: README.md [hunk 5]
```diff
@@ -1,2 +1,81 @@
...
+
+    var axo = new nodeAxosoft(axosoftUrl, credentials);
+
+### Get Work Item (feature)
+
+    //optional parameters
+    var params = {columns: 'name'};
+
+    axo.Features.get(params, function(error, item){
+        console.log(item);
```
Comment (jdgarcia) [inline README.md]: second parameter should be `response`, and the console log should be `response.data` (it's more than 1 item)

File: lib/api.js [hunk 6]
```diff
@@ -12,20 +13,39 @@ var _access_token;
...
-		_access_token = credentials.access_token;
-	} else {
-		_.each(required_credentials, function (credentialName) {
+		_.each(required_password_credentials, function (credentialName) {
 			if (!_.has(credentials, credentialName)) {
 				console.log(credentials);
 				throw new Error('When not providing access_token, you must specify "'
 					+ credentialName + '" within your credentials');
 			}
 		});
```
Comment (jdgarcia) [inline lib/api.js]: unindent this block

File: lib/api.js [hunk 7]
```diff
@@ -12,20 +13,39 @@ var _access_token;
...
 }
 
+function _oauth_flow (credentials) {
+		_.each(required_authorization_code_credentials, function (credentialName) {
+			if (!_.has(credentials, credentialName)) {
+				console.log(credentials);
+				throw new Error('When not providing access_token, you must specify "'
+					+ credentialName + '" within your credentials');
+			}
+		});
```
Comment (jdgarcia) [inline lib/api.js]: unindent this block

File: lib/api.js [hunk 8]
```diff
@@ -40,7 +60,17 @@ function register (base_url, credentials) {
 	}
 
 	_register_base_url(base_url);
-	_register_credentials(credentials);
+	if(credentials.access_token){
+		_access_token = credentials.access_token;
+	} else {
+		if (credentials.grant_type === 'password') {
```
Comment (jdgarcia) [inline lib/api.js]: these 2 lines can be combined into an `else if`

**Comment by jdgarcia** [comment]
It seems weird to have to create an instance of a throwaway NodeAxosoft object just to get access to the `getLoginUrl` method.

What about something like this:
```
var nodeAxosoft = require('node-axosoft');
var loginId = nodeAxosoft.getLoginUrl(url, creds);

// get code

var axo = new nodeAxosoft(url, creds);

**Comment by jdgarcia** [comment]
oh, you don't need to the `new` keyword when getting an object since the main function returns a new object, not an instance of itself.

**Comment by BrettGAxosoft** [comment]
For the Readme stuff, I pretty much copy pastad a bunch of stuff from a node jira wrapper, so I blame them.  Thanks for the notes.

**Comment by jdgarcia** [comment]
@BrettGAxosoft I would still remove `new` from the examples, but everything else looks good.

File: lib/node-axosoft.js [hunk 9]
```diff
@@ -2,12 +2,9 @@ var pkg = require('../package.json');
 
 function NodeAxosoft (base_url, credentials) {
 
-	var api = require('./api');
-
-	api.register(base_url, credentials);
-
 	var node_axosoft = {
 		VERSION: pkg.version,
+		Api: require('./api'),
```
Comment (jdgarcia) [inline lib/node-axosoft.js:7]: actually, since you moved `getLoginUrl`, we don't need to expose `Api` anymore.

Comment (KeifKraken) [reply lib/node-axosoft.js:7]: keeping as it's needed for the access token methods and if we were to make rest calls that don't have methods built into the wrapper

**Pull request merged**